### PR TITLE
AMDGPU: Query MachineModuleInfo from PM instead of MachineFunction

### PR DESCRIPTION
**Pull request by arsenm**

### CODE REVIEW

**Comment by arsenm** [comment]
* **#99679** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/99679?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @arsenm and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: llvm/lib/Target/AMDGPU/SIMemoryLegalizer.cpp [hunk 1]
```diff
@@ -2802,7 +2801,10 @@ bool SIMemoryLegalizer::expandAtomicCmpxchgOrRmw(const SIMemOpInfo &MOI,
 bool SIMemoryLegalizer::runOnMachineFunction(MachineFunction &MF) {
   bool Changed = false;
 
-  SIMemOpAccess MOA(MF);
+  const MachineModuleInfo &MMI =
+      getAnalysis<MachineModuleInfoWrapperPass>().getMMI();
```
Comment (perlfu) [inline llvm/lib/Target/AMDGPU/SIMemoryLegalizer.cpp:2805]: Need to add this to the `AnalysisUsage`?

Comment (arsenm) [reply llvm/lib/Target/AMDGPU/SIMemoryLegalizer.cpp:2805]: It's already in the base MachineFunction getAnalysisUsage

**Review by perlfu — APPROVED** [review]
LGTM

**Pull request merged**